### PR TITLE
Publish snapshots on a schedule, make params bools

### PR DESCRIPTION
**Pull request by kubukoz**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,16 +5,21 @@ on:
       publishSnapshot:
         description: "Publish Snapshot"
         required: false
-        default: "false"
+        default: false
+        type: boolean
```
Comment (kubukoz) [inline .github/workflows/ci.yml:9]: This makes it a checkbox.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -99,7 +104,10 @@ jobs:
   release:
     name: Release
     needs: build
-    if: startsWith(github.ref, 'refs/tags/v') || (github.event_name == 'workflow_dispatch' && github.event.inputs.publishSnapshot == 'true') # || (github.ref == 'refs/heads/main')
+    if:
+      startsWith(github.ref, 'refs/tags/v') ||
+      github.event.inputs.publishSnapshot ||
```
Comment (kubukoz) [inline .github/workflows/ci.yml:109]: note: I removed the `event_name == workflow_dispatch` check, as I think it's redundant - at the moment, inputs can only be set by running a manual dispatch. In other jobs, the input defaults to false.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -135,7 +143,9 @@ jobs:
 
   microsite:
     name: Publish Docs Microsite
-    if: startsWith(github.ref, 'refs/tags/v') || (github.event_name == 'workflow_dispatch' && github.event.inputs.publishMicrosite == 'true')
+    if:
+      startsWith(github.ref, 'refs/tags/v') ||
+      github.event.inputs.publishMicrosite
```
Comment (kubukoz) [inline .github/workflows/ci.yml:148]: same as above, removed the event_name check out of redundancy.

I didn't make the microsite get published on the schedule as schedules only run on the default branch, meaning we couldn't publish the site from `series/0.17` that way.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -99,7 +104,10 @@ jobs:
   release:
     name: Release
     needs: build
-    if: startsWith(github.ref, 'refs/tags/v') || (github.event_name == 'workflow_dispatch' && github.event.inputs.publishSnapshot == 'true') # || (github.ref == 'refs/heads/main')
+    if:
```
Comment (kubukoz) [inline .github/workflows/ci.yml:107]: FYI I tested this out in a smaller repo: https://github.com/kubukoz/demos/actions/runs/5581376805/jobs/10199450921

**Review by Baccata — APPROVED** [review]

**Pull request merged**